src/main/java/practice/StreamPractice.java

```diff
@@ -76,6 +104,10 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * let's write our own impl of Predicate parametrized with Candidate in CandidateValidator.
      */
     public static List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream()
+                .filter(candidate -> new CandidateValidator().test(candidate))
```

Comment:
```suggestion
                .filter(new CandidateValidator())
```
no need to call method test directly, you can just pass object of Predicate impl into method